src/lib/stack-client.js

```diff
@@ -0,0 +1,236 @@
+/* global __TARGET__ */
+/* eslint-env browser */
+
+import CozyClient from 'cozy-client'
+import initializeRealtime from './realtime'
+
+import {
+  ForbiddenException,
+  ServerErrorException,
+  NotFoundException,
+  MethodNotAllowedException,
+  UnavailableStackException,
+  UnavailableSettingsException,
+  UnauthorizedStackException
+} from './exceptions'
+
+import getIcon from './icon'
+
+let cozyClient = undefined
+
+const errorStatuses = {
+  '401': UnauthorizedStackException,
+  '403': ForbiddenException,
+  '404': NotFoundException,
+  '405': MethodNotAllowedException,
+  '500': ServerErrorException
+}
+
+// Utilizy to fetch the resources with cozy-client
+// maintening the compatibility with the legacy 
+// standalone cozy-bar client
+const fetchJSON = function(method, path, body, options={}) {
+  
+  // We mirror here a few lines from cozy-stack-client
+  // because we want a customized fetchJSON
+  const headers = (options.headers = options.headers || {})
+  headers['Accept'] = 'application/json'
+  if (method !== 'GET' && method !== 'HEAD' && body !== undefined) {
+    if (!headers['Content-Type']) {
+      headers['Content-Type'] = 'application/json'
+      body = JSON.stringify(body)
+    }
+  }
+
+  cozyClient.getStackClient().fetch(method, path, body, options).then(
+    resp => {
+      if (typeof errorStatuses[resp.status] === 'function') {
+        throw new errorStatuses[resp.status]()
+      }    
+      const contentType = resp.headers.get('content-type')
+      const isJson = contentType && contentType.indexOf('json') >= 0
+      if (!isJson) {
+        throw new Error("Server response not in JSON")
+      }
+      return resp.json()
+    }
+  )
+}
+
+// Cache a request 
+// with a defaultValue in case of 404
+const withCache(fn, defaultValue) {
+  let cache = undefined;
+  const is404 = function(error) {
```

Comment:
I think this function can be extracted so we don't create a new function on each `withCache` call

Reply:
✅ will do (even if in this case the function will be called only one)